### PR TITLE
added confluence api

### PR DESCRIPTION
**Pull request by liamsegura**
first draft, fixed routing issues, added .env.local example and updated readme for documentation.

solves https://github.com/CrisBado/RandD/issues/12

### CODE REVIEW

File: example .env.local
Comment (matt-greenfield-15) [inline example .env.local:1]: there's a space in the file name?

File: package.json [hunk 1]
```diff
@@ -11,9 +11,11 @@
   },
   "dependencies": {
     "@radix-ui/react-dialog": "^1.0.4",
+    "@types/axios": "^0.14.0",
     "@types/node": "20.5.7",
     "@types/react": "18.2.21",
     "@types/react-dom": "18.2.7",
+    "axios": "^1.5.1",
```
Comment (matt-greenfield-15) [inline package.json]: Can we not just use fetch?

File: src/app/api/confluence/route.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import axios from "axios";
+
+import { NextResponse } from "next/server";
+
+export async function GET(req, res) {
+  try {
+    // Replace 'YOUR_API_TOKEN' with your actual API token
```
Comment (matt-greenfield-15) [inline src/app/api/confluence/route.js]: this isn't relevant anymore

File: src/app/api/confluence/route.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    const cleanedText = textWithoutH3.map((text) =>
+      text.replace(/&ldquo;|&rdquo;/g, "")
+    );
+
+    const cleanedTextObjects = cleanedText.map((text) => ({ text }));
+
+    // Send the modified Confluence API response back to the client
+    console.log(cleanedText);
+
+    return NextResponse.json(cleanedTextObjects);
```
Comment (matt-greenfield-15) [inline src/app/api/confluence/route.js]:
```suggestion
    // Send the modified Confluence API response back to the client
    return NextResponse.json(cleanedTextObjects);
```

File: src/app/api/confluence/route.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  } catch (error) {
+    // Handle errors as needed
+    console.error("Error:", error);
+    return NextResponse.json({ message: "no" });
```
Comment (matt-greenfield-15) [inline src/app/api/confluence/route.js]:
```suggestion
    return NextResponse.status(500).json({ message: error });
```

Comment (matt-greenfield-15) [reply src/app/api/confluence/route.js]: Check what `error` is first though, not sure if it's an object or a string

File: src/app/api/confluence/route.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    const response = await axios.get(confluenceApiUrl, {
+      headers: {
+        "Content-Type": "application/json",
+        username: email,
+        password: apiToken,
+      },
+      auth: {
+        username: email,
+        password: apiToken,
+      },
```
Comment (matt-greenfield-15) [inline src/app/api/confluence/route.js]: Does it need to send them twice like this?

File: src/app/(routes)/page.tsx [hunk 6]
```diff
@@ -45,7 +49,6 @@ export default function Home() {
       </div>
 
       <div className="grid w-full items-center gap-1.5">
-        <p>Showing results for {searchTerm}</p>
```
Comment (matt-greenfield-15) [inline src/app/(routes)/page.tsx]: How come this is going?

File: src/app/(routes)/page.tsx [hunk 7]
```diff
@@ -31,6 +32,9 @@ export default function Home() {
 
   return (
     <main className="flex-col text-center">
+      <Link href="/confluence">
+        <button>Confluence Data here</button>
+      </Link>
```
Comment (matt-greenfield-15) [inline src/app/(routes)/page.tsx]: Buttons inside links isn't valid HTML I'm afraid

File: src/app/(routes)/confluence/page.tsx [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  }, []);
+
+  console.log("confluenceData", confluenceData);
+
+  return (
+    <div>
+      {confluenceData ? (
+        <ul>
+          {confluenceData.map((result, index) => (
+            <li key={index}>
```
Comment (matt-greenfield-15) [inline src/app/(routes)/confluence/page.tsx]: index as a key? Does next lint not complain?

File: src/app/(routes)/page.tsx (hunk 6 above)
Comment (liamsegura) [reply src/app/(routes)/page.tsx]: it was looking at me funny

**Review by matt-greenfield-15 — APPROVED** [review]

**Pull request merged**